### DISCUSSION TITLE
Deploy to Environments only from Tag

### DISCUSSION THREAD

**Discussion by gopisaba** [community/community#64063] [Actions]
### Select Topic Area

Bug

### Body

We have *Selected Deployment Branches* with the regex pattern `v[0-9]+.[0-9]+.[0-9]+` for the *Github Environments* to allow deployments only from the tags but it stopped working recently on all the repositories. Is there any changes on the Github side to make this stop working?

__Error Message__
```Branch "v3.5.0" is not allowed to deploy to env1 due to environment protection rules.```

**Comment by nirav-gajera** [comment]
- Go to the environment settings page.
- Click on the Deployment protection rules tab.
- In the Allowed tags section, add the tags that you want to allow deployments from.
- Click on the Save button.
Once you have updated your deployment branches regex pattern or allowed specific tags, you should be able to deploy from tags again.

**Reply by gopisaba** [reply]
Sorry! can't find anything like Allowed tags section under Deployment protection rules tab in Environment settings page.

<img width="778" alt="Screenshot 2023-08-17 at 14 58 32" src="https://github.com/community/community/assets/12255612/a160c74d-e1ac-4a0e-a739-aabbe91bdd6b">

I set the regex pattern in Deployment branches for the tags but that is no longer missing. What am I missing?

**Comment by javierlga** [comment]
You can't use the `+` quantifier, it sucks but GitHub environment protection uses [fnmatch](https://ruby-doc.org/core-2.5.1/File.html#method-c-fnmatch) from Ruby, reference: [docs](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment#deployment-branches-and-tags), so testing this:
```
puts File.fnmatch('v[0-9]+.[0-9]+.[0-9]+', 'v3.5.0')
```
will always return `false`. Using a wildcard instead  as `puts File.fnmatch('v[0-9]*.[0-9]*.[0-9]*', 'v3.5.0')` will work but it's not super restrictive, as someone could create a tag as the following `v3test.5test.0test` and still be able to deploy, it's a shame it doesn't support the same [filter pattern](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet) as GitHub actions workflows.

**Reply by sassman** [reply]
hope that GitHub at least adds a small help text above the field to elaborate on this counter intuitive detail.
